mpsc-framework/src/main/java/pl/wsztajerowski/MPSCFramework.java

```diff
@@ -0,0 +1,105 @@
+package pl.wsztajerowski;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+
+class MPSCFramework<REQ, RES> implements AutoCloseable {
+
+    private final AtomicReference<InnerBatch<REQ, RES>> currentBatchReference = new AtomicReference<>(new InnerBatch<>());
+    private final ExecutorService consumerExecutor = Executors.newSingleThreadExecutor();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final Consumer<Wrapper<REQ, RES>[]> processor;
+    private final AtomicBoolean consumerAlive = new AtomicBoolean(true);
```

Comment:
volatile will be enough here, atomic might be to heavy for this case, and still with volatile we will have the same semantics